### PR TITLE
[System Integration Tests]: Bootstrap

### PR DESCRIPTION
**Pull request by wavesoft**
This is a preview for the System Integration Tests.

### CODE REVIEW

**Comment by nLight** [comment]
run integration tests

**Comment by wavesoft** [comment]
@mesosphere-mergebot test-it

File: .systemtest.json [hunk 1]
```diff
@@ -0,0 +1,89 @@
+{
```
Comment (wavesoft) [inline .systemtest.json]: An executable `Systemtests` sounds like a good idea 👍

Comment (orlandohohmeier) [inline .systemtest.json]: Aren't they yet? Thought that was the plan...

Comment (wavesoft) [inline .systemtest.json]: Nope, within `system-test.json` you can specify where to find the executables, but having the entire `Systemtests` file executable sounds like a better idea 👍

File: si-tests/_support/index.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
+Cypress.addParentCommand('visitUrl', function (options) {
```
Comment (wavesoft) [inline si-tests/_support/index.js]: Done 👍

File: si-tests/_support/index.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+ */
+Cypress.addChildCommand('contents', function (elements) {
+  return elements.map((index, element) => element.innerText).get();
+});
+
+/**
+ * Extract the string from the given element (including ACE editor) and
+ * check if the parsed JSON equals to the given JSON object.
+ */
+Cypress.addChildCommand('shouldJsonMatch', function (elements, json) {
```
Comment (wavesoft) [inline si-tests/_support/index.js]: Well, it's a `.should` command, and it operates on various elements (including the ACE editor).

Another idea would be to split this in `.aceContents().asJSONObject()`

Comment (orlandohohmeier) [inline si-tests/_support/index.js]: ❤️ `.aceContents().asJSONObject()`

Comment (wavesoft) [inline si-tests/_support/index.js]: Sorry, I have been fighting with cypress in order to create a custom command that returns the contents of the ace editor but it doesn't want to cooperate. I tried promises, `.then` context returns or other hacks but nothing worked out... so for now we have to stick with `shouldJsonMatch` even though it's not the best :(

I am open to suggestions though.

**Comment by wavesoft** [comment]
Closed in favour of #1994

**Pull request closed without merging**